server/portal_server.js

```diff
@@ -4,12 +4,24 @@ const path = require('path');
 const fs = require('fs-extra');
 const bodyParser = require('body-parser');
 const crypto = require('crypto');
+const conf = require('./config.js');
+const stream = require('stream');
+const fetch = require('node-fetch');
+const storage = require('./storage.js');
+
+let isProduction =
```

Comment:
It's confusing that `isProduction` can be `false` when `NODE_ENV=production`. Perhaps naming it something like `useRemoteServices` or `localhostOnly`.